src/commands/serve.rs

```diff
@@ -450,7 +450,17 @@ async fn handle_request(
     match receiver.await {
         Ok(Ok(resp)) => Ok(resp),
         Ok(Err(e)) => Err(e.into()),
-        Err(_) => bail!("guest never invoked `response-outparam::set` method"),
+        Err(_) => {
+            // If the receiver has an error (`RecvError`), it will not describe
+            // the real failure in any meaningful way. Instead we retrieve and
```

Comment:
```suggestion
            // If the receiver has an error (`RecvError`), it only indicates that the task
            // exited before the sender was used, that the sender was dropped
            // before a response was sent. Instead we retrieve and
```

Reply:
I kind of merged our two versions into 5a958a5; hopefully that still captures what you meant in this suggestion!